File: packages/typescript/src/resources/nl-query.ts
import type { HttpClient } from '../http.js';
import type { NLQueryResult } from '../types/query.js';

export class NLQueryResource {
  constructor(private http: HttpClient) {}

  async ask(question: string, opts?: { conversationId?: string; context?: { agent_id?: string; session_id?: string } }): Promise<NLQueryResult> {
    return this.http.post<NLQueryResult>('/v1/nl-query', {
      question,
      conversation_id: opts?.conversationId,
      context: opts?.context,
    });
  }
}

File: packages/python/invariance/resources/nl_query.py
from __future__ import annotations

from typing import Any

from ..http_client import HttpClient
from ..types import NLQueryResult


class NLQueryResource:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    async def ask(
        self,
        question: str,
        *,
        conversation_id: str | None = None,
        context: dict[str, str] | None = None,
    ) -> NLQueryResult:
        body: dict[str, Any] = {"question": question}
        if conversation_id is not None:
            body["conversation_id"] = conversation_id
        if context is not None:
            body["context"] = context
        return await self._http.post("/v1/nl-query", body)
